src/lib/InstantSearch.js

```diff
@@ -47,22 +52,27 @@ class InstantSearch extends EventEmitter {
       searchClient = null,
     } = options;
 
-    if (indexName === null || searchClient === null) {
-      throw new Error(`Usage: instantsearch({
-  indexName: 'indexName',
-  searchClient: algoliasearch('appId', 'apiKey')
-});`);
+    if (indexName === null) {
+      throw new Error(withUsage('The `indexName` option is required.'));
+    }
+
+    if (searchClient === null) {
+      throw new Error(withUsage('The `searchClient` option is required.'));
     }
 
     if (typeof options.urlSync !== 'undefined') {
       throw new Error(
-        'InstantSearch.js V3: `urlSync` option has been removed. You can now use the new `routing` option'
+        withUsage(
+          'The `urlSync` option was removed in InstantSearch.js 3. You may want to use the `routing` option.'
+        )
       );
     }
 
     if (typeof searchClient.search !== 'function') {
       throw new Error(
-        'The search client must implement a `search(requests)` method.'
+        `The \`searchClient\` must implement a \`search\` method.
+
+See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend-search/in-depth/backend-instantsearch/js/`
```

Comment:
How good of an idea is it to include doc links?
I suspect they might 404 one day and it'll be very hard to hunt for those or find out about it. 
I don't have a better solution other than removing them.

Reply:
The alternative (which I already use in the JS client) is to use `alg.li` links (which are also shorter). You can log in using okta to https://short.cm

Reply:
This is what most libraries do.

It is pointless to say that it's a bad usage without pointing to how to actually use it. The user is not expected to know that the `search` method must return a promise for instance.

It is our responsibility to make sure this link doesn't break. To prevent that, I think Algolia has a URL shortener. @Haroenv can give us more details here.